### PR TITLE
ARC: MWDT: add C++ support

### PR DESCRIPTION
**Pull request by evgeniy-paltsev**
Add C++ libs support when building with MWDT toolchain.

### CODE REVIEW

**Review by abrodkin — APPROVED** [review]

File: include/arch/arc/v2/linker.ld [hunk 1]
```diff
@@ -224,6 +224,16 @@ SECTIONS {
 	__kernel_ram_end = .;
 	__kernel_ram_size = __kernel_ram_end - __kernel_ram_start;
 
+#ifdef __MWDT_LINKER_CMD__
+/* mwdt requires _fstack, _estack which will be used in _stkchk.
+ * _stkchk is inserted by mwdt automatically, if _fstack, _estack is not
+ * set correctly, the  brk_s instruction will be called
+ * here, we use a trick to deceive the compiler.
```
Comment (ruuddw) [inline include/arch/arc/v2/linker.ld:231]: Isn't there an option to disable this? I guess not, if you propose this workaround. Can you check with the MetaWare team, or ask for such an option?

**Review by ruuddw — APPROVED** [review]

File: include/arch/arc/v2/linker.ld (hunk 1 above)
Comment (evgeniy-paltsev) [reply include/arch/arc/v2/linker.ld:231]: The only option is to rebuild MWDT libraries with stack check disabled. (well we can rebuild MWDT libraries with buildlib from Zephyr but I really don't like such option :)

I'm going to discuss it with MWDT team so we can probably ship yet another prebuilt library version specially targeted for Zephyr. There are several other things we want to change - I'm collecting the list of changes now.

But anyway that require time and for this Zephyr release we wave to use workarounds.

**Comment by cfriedt** [comment]
@evgeniy-paltsev - is this ready to merge?

**Review by stephanosio — APPROVED** [review]
Looks ok to me.

**Review by alexanderwachter — APPROVED** [review]

**Comment by evgeniy-paltsev** [comment]
> evgeniy-paltsev - is this ready to merge?

Yes, it's ready.

**Pull request merged**